### PR TITLE
Update public DNS records in COOL-hosted zone

### LINKED ISSUE #256: DNS No Longer Updates After Migration of DNS Management to COOL
# 🐛 Bug Report

DNS applies no longer work after the DNS management migration to the COOL account.

## To Reproduce

- Perform an apply that updates DNS.
- Watch DNS updates not work.

## Expected behavior

DNS updates work.

@dav3r and I had to manually update the DNS records in the COOL account to reflect new instance IPs after doing a redeployment of the bastion instances for CyHy and BOD.

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR switches this project from publishing public DNS records to a zone in our "CyHy Production" account to a zone in [the COOL](https://github.com/cisagov/cool-system).

Resolves #256.

## 💭 Motivation and Context
We recently migrated the public zone for `cyber.dhs.gov` to the COOL, so this change is required in order for bastion (`cyhy`, `bod`, `mgmt`) DNS entries to be made publicly available.

## 🧪 Testing
I tested this by successfully applying these Terraform changes in our current Production workspace.  Note that I had to do a few things before this apply would work:
* Ensure that I had the newly-required `cool-dns-route53resourcechange-cyber.dhs.gov` and `cool-terraform-readstate` profiles set up in my AWS credentials file.
* I manually removed the existing bastion Terraform state entries (`terraform state rm aws_route53_record.cyhy_bastion_pub_A`, `terraform state rm aws_route53_record.bod_bastion_pub_A`) because the new COOL DNS provider isn't allowed to access the old CyHy account DNS zone.
* I manually deleted the two bastion A records that were in the COOL DNS zone because Terraform couldn't overwrite them when it attempted to apply them.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -3,11 +3,22 @@ provider "aws" {
   region = var.aws_region
 }
 
+# This is the provider that can make changes to DNS entries
+# in the public cyber.dhs.gov zone.
+#
+# NOTE: After the CyHy AWS account moves in to the COOL, it will be possible
```
Comment (jsf9k) [inline terraform/main.tf]: in to --> into

**Review by jsf9k — APPROVED** [review]
Nice work!

File: terraform/main.tf (hunk 1 above)
Comment (dav3r) [reply terraform/main.tf]: Thanks- see https://github.com/cisagov/cyhy_amis/pull/257/commits/6032cc87d1a934398e3964823253f067e81c30cc.

**Review by mcdonnnj — APPROVED** [review]
Strong work 💪 Thanks for taking care of this issue.

**Pull request merged**